include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -775,7 +776,7 @@ inline void format_bam::write_alignment_record([[maybe_unused]] stream_type &  s
             /* mapq        */ mapq,
             /* bin         */ reg2bin(ref_offset.value_or(-1), ref_length),
             /* n_cigar_op  */ static_cast<uint16_t>(cigar_vector.size()),
-            /* flag        */ flag,
+            /* flag        */ static_cast<uint16_t>(flag),
```

Comment:
the core should not hold a sam_flag type because the entire alignment record core is loaded as one binary dump?
Or is the field also read separately?

Reply:
Yes, the entire alignment record core is loaded as one binary dump!